src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/RemoveOrganizationUserCommand.cs

```diff
@@ -34,123 +47,107 @@ public RemoveOrganizationUserCommand(
         _pushRegistrationService = pushRegistrationService;
         _currentContext = currentContext;
         _hasConfirmedOwnersExceptQuery = hasConfirmedOwnersExceptQuery;
+        _getOrganizationUsersManagementStatusQuery = getOrganizationUsersManagementStatusQuery;
+        _featureService = featureService;
+        _timeProvider = timeProvider;
     }
 
-    public async Task RemoveUserAsync(Guid organizationId, Guid organizationUserId, Guid? deletingUserId)
+    public async Task RemoveUserAsync(Guid organizationId, Guid userId)
     {
-        var organizationUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
+        var organizationUser = await _organizationUserRepository.GetByOrganizationAsync(organizationId, userId);
         ValidateDeleteUser(organizationId, organizationUser);
 
-        await RepositoryDeleteUserAsync(organizationUser, deletingUserId);
+        await RepositoryDeleteUserAsync(organizationUser, null, null);
 
         await _eventService.LogOrganizationUserEventAsync(organizationUser, EventType.OrganizationUser_Removed);
     }
 
-    public async Task RemoveUserAsync(Guid organizationId, Guid organizationUserId, EventSystemUser eventSystemUser)
+    public async Task RemoveUserAsync(Guid organizationId, Guid organizationUserId, Guid? deletingUserId)
     {
         var organizationUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
         ValidateDeleteUser(organizationId, organizationUser);
 
-        await RepositoryDeleteUserAsync(organizationUser, null);
+        await RepositoryDeleteUserAsync(organizationUser, deletingUserId, null);
 
-        await _eventService.LogOrganizationUserEventAsync(organizationUser, EventType.OrganizationUser_Removed, eventSystemUser);
+        await _eventService.LogOrganizationUserEventAsync(organizationUser, EventType.OrganizationUser_Removed);
     }
 
-    public async Task RemoveUserAsync(Guid organizationId, Guid userId)
+    public async Task RemoveUserAsync(Guid organizationId, Guid organizationUserId, EventSystemUser eventSystemUser)
     {
-        var organizationUser = await _organizationUserRepository.GetByOrganizationAsync(organizationId, userId);
+        var organizationUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
         ValidateDeleteUser(organizationId, organizationUser);
 
-        await RepositoryDeleteUserAsync(organizationUser, null);
+        await RepositoryDeleteUserAsync(organizationUser, null, eventSystemUser);
```

Comment:
Please use a named argument when passing in `null`s to these private methods to avoid mixups, e.g. `await RepositoryDeleteUserAsync(organizationUser, deletingUserId: null, eventSystemUser);`